### PR TITLE
Add Beacon BlockState Support

### PR DESCRIPTION
**Pull request by Fusezion**
List of everything added is below

- Added ExprBeaconTier
- Added ExprBeaconEffect
- Added ExprBeaconEffectRange
- Added ExprBeaconEntitiesInRange
- ~~Added BeaconTier ENUM~~
- ~~Add beacontier into english.lang~~
- ~~Registered ClassInfo for BeaconTier ENUM~~

File changes to PaperEvents was done to reorganize the file into how they are labeled, we had a few player events at the end of the file
- Added PlayerChangeBeaconEffectEvent to PaperEvents
- Added BeaconDeactivatedEvent to PaperEvents
- Added BeaconActivatedEvent to PaperEvents

As always I probably missed a period or colon hell I might have even missed some annotations
suggestion on some syntax is welcomed as some I have `beacon` optional and most I do not

### CODE REVIEW

File: src/main/java/com/shanebeestudios/skbee/elements/other/events/PaperEvents.java [hunk 1]
```diff
@@ -272,72 +344,40 @@ public Player get(AnvilDamagedEvent event) {
                 public Player get(BeaconEffectEvent e) {
                     return e.getPlayer();
                 }
-            }, 0);
+            }, EventValues.TIME_NOW);
             // TODO These two values will make more sense in the future (Currently have a PR for potion effects in Skript)
             EventValues.registerEventValue(BeaconEffectEvent.class, PotionEffectType.class, new Getter<>() {
                 @Override
```
Comment (Fusezion) [inline src/main/java/com/shanebeestudios/skbee/elements/other/events/PaperEvents.java:350]: btw this `TODO` do we need this anymore? I'm guessing this it related to before skript added PotionEffect and PotionEffectType in 2.5 or whatever it was

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprBeaconEffect.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import org.bukkit.block.Beacon;
+import org.bukkit.block.Block;
+import org.bukkit.event.Event;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Beacon - Primary/Secondary Effect")
+@Description({"The primary/secondary potion effect of a beacon.",
+        "Note:",
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprBeaconEffect.java:22]: probably best to put this on another line
```suggestion
        "\nNote:",
```

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprBeaconEffect.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        "\nWhen getting the primary/secondary potion effect it will return as a Potion Effect and not a Potion Effect Type.",
+        "\nIf you update the beacon inventory by opening it and placing an item you will lose any custom effects not normally applicable."})
+@Examples({"on right click on a beacon with a diamond:",
+        "broadcast \"Primary: %primary beacon effect of clicked block%\"",
+        "broadcast \"Secondary: %secondary beacon effect of clicked block%\"",
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprBeaconEffect.java:27]:
```suggestion
        "\tbroadcast \"Primary: %primary beacon effect of clicked block%\"",
        "\tbroadcast \"Secondary: %secondary beacon effect of clicked block%\"",
```

File: src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprBeaconEntitiesInRange.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import org.bukkit.block.Beacon;
+import org.bukkit.block.Block;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Name("Beacon - Entities in effect range")
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/other/expressions/ExprBeaconEntitiesInRange.java:25]:
```suggestion
@Name("Beacon - Entities in Effect Range")
```

**Review by ShaneBeee — CHANGES_REQUESTED** [review]
Just a few minor things

**Pull request merged**